source/particle/property/crystal_preferred_orientation.cc

```diff
@@ -672,16 +678,54 @@ namespace aspect
         const Tensor<2,3> strain_rate_nondimensional = nondimensionalization_value != 0 ? strain_rate_3d/nondimensionalization_value : strain_rate_3d;
         const Tensor<2,3> velocity_gradient_tensor_nondimensional = nondimensionalization_value != 0 ? velocity_gradient_tensor/nondimensionalization_value : velocity_gradient_tensor;
 
-        // create output variables
+        // Create output variables
         std::vector<double> deriv_volume_fractions(n_grains);
         std::vector<Tensor<2,3>> deriv_a_cosine_matrices(n_grains);
 
-        // create shortcuts
+        // Create shortcuts
         const std::array<double, 4> &tau = ref_resolved_shear_stress;
-
         std::vector<double> strain_energy(n_grains);
         double mean_strain_energy = 0;
 
+        // First initiate the slip_direction_reference (vector l) in Kaminski 2001) and
+        // slip_normal_reference (vector n) for olivine A,B,C,E types
+        std::array<Tensor<1,3>,4> slip_normal_reference {{Tensor<1,3>({0,1,0}),Tensor<1,3>({0,0,1}),Tensor<1,3>({0,1,0}),Tensor<1,3>({1,0,0})}};
+        std::array<Tensor<1,3>,4> slip_direction_reference {{Tensor<1,3>({1,0,0}),Tensor<1,3>({1,0,0}),Tensor<1,3>({0,0,1}),Tensor<1,3>({0,0,1})}};
+
```

Comment:
Please package this code into:
```suggestion
if (deformation_type == DeformationType::olivine || deformation_type == DeformationType::enstatite)
{
        // slip_normal_reference (vector n) for olivine A,B,C,E types
        std::array<Tensor<1,3>,4> slip_normal_reference {{Tensor<1,3>({0,1,0}),Tensor<1,3>({0,0,1}),Tensor<1,3>({0,1,0}),Tensor<1,3>({1,0,0})}};
        std::array<Tensor<1,3>,4> slip_direction_reference {{Tensor<1,3>({1,0,0}),Tensor<1,3>({1,0,0}),Tensor<1,3>({0,0,1}),Tensor<1,3>({0,0,1})}};
}
else if (deformation_type == DeformationType::clinopyroxene)
{
...
}
else
  AssertThrow(false, ExcMessage("Unknown deformation type"));
```

Reply:
I guess we still need to have the two std::array<Tensor<1,3>,4> initiated in the beginning outside of the if statement so that their value won't vanish after the if blocks?